package.json

```diff
@@ -23,7 +23,8 @@
   },
   "scripts": {
     "build": "rollup -c",
-    "test": "npm run build && tap test/*.js && jsmd readme.md",
+    "test": "npm run build && tap test/*.js && jsmd readme.md && npm run test:typescript",
+    "test:typescript": "node ./node_modules/typescript/lib/tsc.js --noEmit test/typescript.ts",
```

Comment:
Potentially just `tsc --noEmit test/typescript.ts` would be cleaner here.

Reply:
Good point, the pull request description says "the TS test invokes `tsc.js` directly since `typescript/bin/tsc` will not run on Windows", but in my experience npm scripts have worked fine on Windows

Reply:
I am on Windows 10 and have TS 3.0.3 installed globally (which is recommended if you do TS dev regularly).

Calling `tsc` directly invokes my global install. I think if I had no global install then nothing would happen. Calling `node_modules/bin/tsc` doesn't work because [`tsc` is a *nix shell script](https://github.com/Microsoft/TypeScript/blob/master/bin/tsc), not an npm script. So to ensure we call the project-local tsc and it works cross-platform, I did:
`node ./node_modules/typescript/lib/tsc.js`

Reply:
`npm run` scripts use your locally-installed versions of modules by default, so `tsc` works just fine here.

It's not actually a unix script (Windows is cool with the `#!` at the top, it just ignores it).

npm automatically launches the [script specified in the package.json](https://github.com/Microsoft/TypeScript/blob/d2647a1ddaf270558c6a55bd14a22d80e2b6df1f/package.json#L25 with node when you use it in a npm run script.

Reply:
@msssk so the reasoning behind 2.0 instead of 3.0 is that it means projects still on old versions of typescript can use this project, so going forward in the future we should use the oldest version of typescript that has the feature set we need?

Reply:
@macdja38 correct.

Reply:
Did some more testing and updated `package.json` to set TS specifically to 2.2.2. 2.0.0 does not have `Partial` which is very helpful for our typedefs.